### PR TITLE
Add new `exclusion` ad slot to all pages

### PR DESCRIPTION
**Pull request by simonbyford**
## What does this change?

Adds a new `exclusion` ad slot to the header of the page, just before `top-above-nav`.

## Why?

AdOps would like to be able to target _exclusion_ campaigns against certain breaking news pages in order to disable Opt Out advertising on those pages.

To get this working, Opt Out have asked us to introduce a new `exclusion` ad slot. When it is filled, Opt Out's new logic will magically prevent all other ads making their way onto the page.

URL targeting will determine whether or not this slot is filled. Something like the following:

Screenshot 2022-12-02 at 12 45 54

The new `exclusion` slot isn't visible to the user and will be ignored by GAM.

## Next steps

1) ~~Replicate this change on frontend so it works on non-DCR rendered pages (like fronts)~~ Update: https://github.com/guardian/frontend/pull/25732

2) ~~Configure the `exclusion` ad slot in the [Opt Out console](https://dashboard.optoutadvertising.com/adslottags/publisher/33)~~ Update: done!

Screenshot 2022-12-02 at 13 59 45

3) We _might_ need to add some logic [around here](https://github.com/guardian/frontend/blob/0d2568f996e1f0fc5068f6217169b34c501d7a97/static/src/javascripts/projects/commercial/modules/consentless/define-slot.ts#L31-L34) to tidy up empty ad slots, but this also may just work

## Screenshots

Here is a sample request to Opt Out with the `exclusion` ad slot in place:

Screenshot 2022-12-01 at 16 58 10

### CODE REVIEW

File: dotcom-rendering/src/web/components/AdSlot.tsx [hunk 1]
```diff
@@ -476,6 +476,16 @@ export const AdSlot = ({
 				/>
 			);
 		}
+		case 'exclusion': {
```
Comment (arelra) [inline dotcom-rendering/src/web/components/AdSlot.tsx:479]: Minor - is there a way to make it clear this is for consentless only. Either naming or a comment?

**Review by arelra — APPROVED** [review]
👍

File: dotcom-rendering/src/web/components/AdSlot.tsx (hunk 1 above)
Comment (simonbyford) [reply dotcom-rendering/src/web/components/AdSlot.tsx:479]: Good shout, I will add!

**Comment by mchv** [comment]
@simonbyford @SiAdcock I think we may have high number of false positive sentry reported errors linked to this

Screenshot 2022-12-15 at 09 47 47
Screenshot 2022-12-15 at 09 47 30

I think a way to fix this could be the following code in [create-advert.ts](https://github.com/guardian/frontend/blob/main/static/src/javascripts/projects/commercial/modules/dfp/create-advert.ts#L6-L29)

```typescript
const createAdvert = (
	adSlot: HTMLElement,
	additionalSizes?: SizeMapping,
): Advert | null => {
	try {
		const advert = new Advert(adSlot, additionalSizes);
		return advert;
	} catch {
		const errMsg = `Could not create advert. Ad slot: ${
			adSlot.id
		}. Additional Sizes: ${JSON.stringify(additionalSizes)}`;
		if (adSlot.id == 'dfp-ad--exclusion) {
		  // do not log or report an error, we expect the ad slot to fail for the moment while integrating opt-out
		} else {
		    log('commercial', errMsg);
		    reportError(
			new Error(errMsg),
			{
				feature: 'commercial',
			},
			false,
		    );
                 }

		return null;
	}
};
```

**Comment by mchv** [comment]
@SiAdcock just let me know, that this should be properly fixed now through https://github.com/guardian/commercial-core/pull/734

**Pull request merged**